be/src/exprs/vectorized/cast_expr.cpp

```diff
@@ -1423,6 +1446,24 @@ Expr* VectorizedCastExprFactory::from_thrift(const TExprNode& node, bool allow_t
             return new VectorizedCastExpr<TYPE_VARCHAR, TYPE_HLL, false>(node);
         }
     }
+    // Cast string to array<ANY>
+    if (from_type == TYPE_VARCHAR && to_type == TYPE_ARRAY) {
+        TypeDescriptor cast_to = TypeDescriptor::from_thrift(node.type);
+        TExprNode cast;
+        cast.type = cast_to.children[0].to_thrift();
```

Comment:
why not set array element type info in FE

Reply:
element type is set into `children` for complex types.